website_menu_by_user_status/controllers/main.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.web.controllers.main import Home
+
+
+class Website(Home):
+
+    def _get_pages(self):
+        """
+        All pages width url
+        """
+        return request.env['website.menu'].sudo().search([]).filtered('url')
+
+    def _check_user_access(self):
+        """
+        Is user logged
+        """
+        return request.env.user == request.website.user_id or False
+
+    def _get_suggest_page(self):
+        """
+        All pages accessible by the user (logged or not)
+        """
+        if self._check_user_access():
```

Comment:
BTW couldn't this be achieved with some record rule?

Reply:
@simahawk I think it's possible but this is little hard to check if a user is logged in or not. Do you have an idea to do this ?

Reply:
I have a PoC working w/ groups + rule. I think this module should be based upon this and get rid of the check on pages:  adds logic that is not perfomant and goes far beyond the purpose of the module. Page/routes permission should be granted by routes/pages logic themselves. Very likely we gonna get rid of the redirect issue too.

Can't work on this today but I'm going to get back w/ another PR most probably on Monday.

Reply:
there you go https://github.com/OCA/website/pull/394